src/super_gradients/training/losses/yolox_loss.py

```diff
@@ -0,0 +1,405 @@
+        self.strides = strides
+        self.im_size = im_size
+        self.num_classes = num_classes
+
+        self.center_sampling_radius = center_sampling_radius
+        self.use_l1 = use_l1
+        self.l1_loss = nn.L1Loss(reduction="none")
+        self.bcewithlog_loss = nn.BCEWithLogitsLoss(reduction="none")
+        self.iou_loss = IOUloss(reduction="none")
+
+    def forward(self, model_output, targets):
+        if isinstance(model_output, tuple) and len(model_output) == 2:
+            # in test/eval mode the Yolo v5 model output a tuple where the second item is the raw predictions
+            _, predictions = model_output
+        else:
+            predictions = model_output
+
+        return self.compute_loss(predictions, targets)
+
+    @staticmethod
+    def _make_grid(nx=20, ny=20):
+        yv, xv = torch.meshgrid([torch.arange(ny), torch.arange(nx)])
+        return torch.stack((xv, yv), 2).view((1, 1, ny, nx, 2)).float()
+
+    def compute_loss(self, predictions: List[torch.Tensor], targets: torch.Tensor) \
+            -> Tuple[torch.Tensor, torch.Tensor]:
+        x_shifts, y_shifts, expanded_strides, transformed_outputs, raw_outputs = self.prepare_predictions(predictions)
+
+        bbox_preds = transformed_outputs[:, :, :4]  # [batch, n_anchors_all, 4]
+        obj_preds = transformed_outputs[:, :, 4:5]  # [batch, n_anchors_all, 1]
+        cls_preds = transformed_outputs[:, :, 5:]  # [batch, n_anchors_all, n_cls]
+
+        # calculate targets
+        total_num_anchors = transformed_outputs.shape[1]
+        cls_targets = []
+        reg_targets = []
+        l1_targets = []
+        obj_targets = []
+        fg_masks = []
+
+        num_fg, num_gts = 0., 0.
+
+        for batch_idx in range(transformed_outputs.shape[0]):
+            labels_im = targets[targets[:, 0] == batch_idx]
+            num_gt = labels_im.shape[0]
+            num_gts += num_gt
+            if num_gt == 0:
+                cls_target = transformed_outputs.new_zeros((0, self.num_classes))
+                reg_target = transformed_outputs.new_zeros((0, 4))
+                l1_target = transformed_outputs.new_zeros((0, 4))
+                obj_target = transformed_outputs.new_zeros((total_num_anchors, 1))
+                fg_mask = transformed_outputs.new_zeros(total_num_anchors).bool()
+            else:
+                gt_bboxes_per_image = labels_im[:, 2:6] * self.im_size
+                gt_classes = labels_im[:, 1]
+                bboxes_preds_per_image = bbox_preds[batch_idx]
+
+                try:
+                    gt_matched_classes, fg_mask, pred_ious_this_matching, matched_gt_inds, num_fg_img = \
+                        self.get_assignments(batch_idx, num_gt, total_num_anchors, gt_bboxes_per_image,
+                                             gt_classes, bboxes_preds_per_image,
+                                             expanded_strides, x_shifts, y_shifts, cls_preds, obj_preds)
+                except RuntimeError:
```

Comment:
Do we want to keep this? It's supposed to help when during training it goes OOM while trying to do the assignment. I actually saw it happen a few times